### PR TITLE
[WB-1799] Nested Configs

### PR DESCRIPTION
**Pull request by hu-po**
Modifies the `jsonschema` for sweeps to allow for nested parameters.

### Addresses:
- [WB-1799](https://wandb.atlassian.net/browse/WB-1799)

### Features
- Nested parameter functionality is added through the new parameter type `param-dict`. Parameters of this type can contain any other parameter type, including another `param-dict`.

### CODE REVIEW

File: src/sweeps/config/schema.py [hunk 1]
```diff
@@ -86,6 +86,10 @@ def fill_parameter(parameter_name: str, config: Dict) -> Optional[Tuple[str, Dic
         except jsonschema.ValidationError:
             continue
         else:
+            if schema_name == "param_dict":
+                # Don't try and fill default values for these
+                # just pass the config forward.
+                return schema_name, config
```
Comment (dannygoldstein) [inline src/sweeps/config/schema.py:92]: can we just raise a custom exception here for type = "param_dict" and then handle it on line 385 of _unnest. that way we dont have to implement stubs TYPE=param_dict in HyperParemeter

File: src/sweeps/params.py [hunk 2]
```diff
@@ -354,17 +371,71 @@ def from_config(cls, config: Dict):
...
+                                _d: Dict = dict()
+                                subdict[subkey] = _d
+                                subdict = _d
+                        if isinstance(subdict, dict):
+                            subdict[subkeys[-1]] = d.pop(k)
+
+        # Add only the hyperparameters which aren't dicts
+        config: Dict = dict()
+        for param in self:
+            if param.type != HyperParameter.DICT:
```
Comment (dannygoldstein) [inline src/sweeps/params.py:430]: could this condition ever be true, at this stage?

**Review by dannygoldstein — COMMENTED** [review]
looking good ! just a few comments below and requests for more tests, and i think we should be good to go :) 

please add a test where a nested parameter is specified without all fields filled, and check that the default filler works properly, example

`{"a": {"min": 0, "max": 1}}` should fill to `{"a": {"min": 0, "max": 1, "distribution": "int_uniform"}}`

File: src/sweeps/config/schema.py (hunk 1 above)
Comment (hu-po) [reply src/sweeps/config/schema.py:92]: thats a pretty good idea, makes for less lines of code too

**Review by dannygoldstein — APPROVED** [review]

**Pull request merged**